### ISSUE TITLE
How to use zap.logger(not sugar) with kit.logger?

### ISSUE THREAD

**Issue by MaruHyl** [go-kit/kit#854]
I love kit. This is what I want.
But how to use zap.logger(not sugar) with kit.logger?
For examle, in zap
```
Logger.Info("msg", zap.String("A", A), zap.Duration("B", B))
```

**Comment by ChrisHines** [comment]
The Go kit `log` package expects type information encoded in Go `interface{}` values, not `zap.Field` values. You will need to adapt one method to the other for your use case.

Can you provide more information about what you are trying to do? Do you want an implementation of the Go kit `log.Logger` interface that writes log data using the `zap` package, or do you have code that creates log events with the `zap` API and you want it to write log data with a Go kit `log.Logger`?

**Comment by MaruHyl** [comment]
@ChrisHines I write a adapter for kit.Logger with zap.Logger
```

type logFunc func(kv ...interface{}) error

func (f logFunc) Log(kv ...interface{}) error {
	return f(kv)
}

func ZapAdapter(logger *zap.Logger, level zapcore.Level) log.Logger {
	var realLogFunc func(msg string, keysAndValues ...interface{})
	switch level {
	case zapcore.DebugLevel:
		realLogFunc = logger.Sugar().Debugw
	case zapcore.InfoLevel:
		realLogFunc = logger.Sugar().Infow
	case zapcore.WarnLevel:
		realLogFunc = logger.Sugar().Warnw
	case zapcore.ErrorLevel:
		realLogFunc = logger.Sugar().Errorw
	case zapcore.DPanicLevel:
		realLogFunc = logger.Sugar().DPanicw
	case zapcore.PanicLevel:
		realLogFunc = logger.Sugar().Panicw
	case zapcore.FatalLevel:
		realLogFunc = logger.Sugar().Fatalw
	}
	return logFunc(func(kv ...interface{}) error {
		realLogFunc("", kv...)
		return nil
	})
}
```
Can I make this a PR?

**Comment by MaruHyl** [comment]
Use like this
```
func NewGRPCServer(endpoints endpoint.Set, logger *zap.Logger) dataservice.DataServiceServer {
	options := []grpctransport.ServerOption{
		grpctransport.ServerErrorLogger(kit.ZapAdapter(logger, zapcore.ErrorLevel)),
	}
        //.....
}
```
So I can catch `grpctransport.ServerErrorLog` to my zap.Logger

**Comment by ChrisHines** [comment]
@MaruHyl I'm fine with a PR for this. It should go into a new package in a subdirectory of the log package. I recommend github.com/go-kit/kit/log/zap as the import path. Follow the example of package github.com/go-kit/kit/log/logrus. We'll want these type of adapters to have a similar API to each other as much as possible.

**Comment by ChrisHines** [comment]
Closed by #855.
